tests/agent/test_docker_agent.py

```diff
@@ -476,9 +476,20 @@ def test_docker_agent_deploy_flow_run_config(api, run_kind, has_docker_storage):
 
     if run_kind == "docker":
         env = {"TESTING": "VALUE"}
-        run = DockerRun(image=image, env=env)
+        host_config = {"shm_size": "128m"}
```

Comment:
Can you include `auto_remove=False` here so we can confirm it overrides correctly as well?

Reply:
Yes I can. Should I add it in with `pytest.mark.parametrize` so that we  can test both of them fairly easily in the same test?

Reply:
Since we also set `extra_hosts` we can cover both overriding an existing value and leaving one untouched without any parameterization here